app/routes/tags/$tag.tsx

```diff
@@ -0,0 +1,124 @@
+import {useState, ReactNode} from 'react'
+import {LoaderFunction} from '@remix-run/cloudflare'
+import {reloadInBackgroundIfNeeded} from '~/server-utils/kv-cache'
+import {loadTag, Tag as TagType} from '~/server-utils/stampy'
+import Dialog from '~/components/dialog'
+
+type Props = {
+  tags: string[]
+  selectQuestion: (pageid: string, title: string) => void
+  [k: string]: any
+}
+
+export const loader = async ({request, params}: Parameters<LoaderFunction>[0]) => {
+  const {tag} = params
+  if (!tag) {
+    throw Error('missing tag name')
+  }
+
+  try {
+    return await loadTag(request, tag)
+  } catch (error: any) {
+    console.error(`error fetching tag "${tag}":`, error)
+    return {
+      error: error.toString(),
+      timestamp: new Date().toISOString(),
+      data: [],
+    }
+  }
+}
+
+export async function fetchTag(tagName: string): Promise<TagType | never[]> {
+  const url = `/tags/${encodeURIComponent(tagName)}`
+  return fetch(url).then(async (response) => {
+    const json: Awaited<ReturnType<typeof loader>> = await response.json()
+    if ('error' in json) console.error(json.error)
+    const {data, timestamp} = json
+
+    reloadInBackgroundIfNeeded(url, timestamp)
+
+    return data
+  })
+}
+
+function TagQuestions({
+  tag,
+  selectQuestion,
+  clearTag,
+}: {
+  tag: TagType
+  selectQuestion: (pageid: string, title: string) => void
+  clearTag: () => void
+}) {
+  const TagDialog = ({children}: {children: ReactNode | ReactNode[]}) => (
+    <Dialog onClose={() => clearTag()}>
+      <div className="dialog-title">
+        Select a question from the <b>{tag.name}</b> tag...
+      </div>
+      {children}
+    </Dialog>
+  )
+
+  if (!tag.rowId) {
+    return (
+      <TagDialog>
+        <div className="loader"></div>
```

Comment:
display a loader thingy until the tag's questions have been fetched

Reply:
if I understand correctly, we always display question tags, but then lazy load the other questions for a given tag if you click the tag?

that sounds like correct behaviour to me, no unnecessary loading for people not interested in tags, but no loader when opening just the question itself 👍

Reply:
exactly. It looks like this:
<img width="774" alt="image" src="https://user-images.githubusercontent.com/3942390/219963403-bb66fb5d-042a-4832-a578-c5e291d822d6.png">